pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/UpsertTableSegmentUploadIntegrationTest.java

```diff
@@ -87,7 +88,24 @@ public void setUp()
   @AfterClass
   public void tearDown()
       throws IOException {
-    dropRealtimeTable(getTableName());
+    String realtimeTableName = TableNameBuilder.REALTIME.tableNameWithType(getTableName());
+
+    // Test dropping all segments one by one
+    List<String> segments = listSegments(realtimeTableName);
+    assertFalse(segments.isEmpty());
+    for (String segment : segments) {
+      dropSegment(realtimeTableName, segment);
+    }
+    // NOTE: There is a delay to remove the segment from property store
+    TestUtils.waitForCondition((aVoid) -> {
+      try {
+        return listSegments(realtimeTableName).isEmpty();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }, 60_000L, "Failed to drop the segments");
```

Comment:
nit, why is this test during tearDown?

Reply:
This is part of the tear-down process. `OfflineClusterIntegrationTest` has similar test. We may consider adding dependency to the test so that we can move it into a test, but prefer doing it separately

Reply:
yeah good enough for now we can follow up